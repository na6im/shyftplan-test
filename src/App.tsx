import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from 'react-query';
import EventsScreen from './screens/EventsScreen';

const queryClient = new QueryClient();

function App() {
  return (
    <QueryClientProvider client={queryClient}>
      <div>
        <BrowserRouter>
          <Route path="/" exact component={EventsScreen} />
        </BrowserRouter>
      </div>
    </QueryClientProvider>
  );
}

export default App;
