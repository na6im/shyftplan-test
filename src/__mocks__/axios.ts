import axios from 'axios';

const mockAxios = jest.genMockFromModule<typeof axios>('axios');
mockAxios.create = jest.fn(() => mockAxios) as any;

export default mockAxios;
