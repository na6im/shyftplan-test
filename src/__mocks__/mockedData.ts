export const eventsData = {
  items: [
    {
      id: 1,
      position: { name: 'Books', color: 'lime', id: 83 },
      startsAt: '2020-08-01T06:00:00.000Z',
      endsAt: '2020-08-01T11:00:00.000Z',
    },
    {
      id: 48,
      position: { name: 'Health', color: 'yellow', id: 72 },
      startsAt: '2020-09-10T06:00:00.000Z',
      endsAt: '2020-09-10T14:00:00.000Z',
    },
    {
      id: 61,
      position: { name: 'Sports', color: 'lime', id: 233 },
      startsAt: '2020-12-24T10:00:00.000Z',
      endsAt: '2020-12-24T18:00:00.000Z',
    },
    {
      id: 95,
      position: { name: 'Sports', color: 'lime', id: 45 },
      startsAt: '2020-06-09T09:00:00.000Z',
      endsAt: '2020-06-09T13:00:00.000Z',
    },
    {
      id: 106,
      position: { name: 'Books', color: 'lime', id: 83 },
      startsAt: '2020-04-22T07:00:00.000Z',
      endsAt: '2020-04-22T13:00:00.000Z',
    },
    {
      id: 134,
      position: { name: 'Health', color: 'yellow', id: 72 },
      startsAt: '2020-01-24T10:00:00.000Z',
      endsAt: '2020-01-24T15:00:00.000Z',
    },
    {
      id: 166,
      position: { name: 'Sports', color: 'lime', id: 45 },
      startsAt: '2020-10-06T07:00:00.000Z',
      endsAt: '2020-10-06T13:00:00.000Z',
    },
    {
      id: 166,
      position: { name: 'Sports', color: 'lime', id: 233 },
      startsAt: '2020-07-14T10:00:00.000Z',
      endsAt: '2020-07-14T14:00:00.000Z',
    },
    {
      id: 195,
      position: { name: 'Sports', color: 'lime', id: 45 },
      startsAt: '2020-11-10T07:00:00.000Z',
      endsAt: '2020-11-10T13:00:00.000Z',
    },
    {
      id: 239,
      position: { name: 'Sports', color: 'lime', id: 45 },
      startsAt: '2020-02-05T06:00:00.000Z',
      endsAt: '2020-02-05T13:00:00.000Z',
    },
  ],
  pagination: { offset: 0, limit: 10, count: 50 },
};

export const eventsDataNoMore = {
  items: [
    {
      id: 1,
      position: { name: 'Books', color: 'lime', id: 83 },
      startsAt: '2020-08-01T06:00:00.000Z',
      endsAt: '2020-08-01T11:00:00.000Z',
    },
    {
      id: 48,
      position: { name: 'Health', color: 'yellow', id: 72 },
      startsAt: '2020-09-10T06:00:00.000Z',
      endsAt: '2020-09-10T14:00:00.000Z',
    },
    {
      id: 61,
      position: { name: 'Sports', color: 'lime', id: 233 },
      startsAt: '2020-12-24T10:00:00.000Z',
      endsAt: '2020-12-24T18:00:00.000Z',
    },
    {
      id: 95,
      position: { name: 'Sports', color: 'lime', id: 45 },
      startsAt: '2020-06-09T09:00:00.000Z',
      endsAt: '2020-06-09T13:00:00.000Z',
    },
    {
      id: 106,
      position: { name: 'Books', color: 'lime', id: 83 },
      startsAt: '2020-04-22T07:00:00.000Z',
      endsAt: '2020-04-22T13:00:00.000Z',
    },
    {
      id: 134,
      position: { name: 'Health', color: 'yellow', id: 72 },
      startsAt: '2020-01-24T10:00:00.000Z',
      endsAt: '2020-01-24T15:00:00.000Z',
    },
    {
      id: 166,
      position: { name: 'Sports', color: 'lime', id: 45 },
      startsAt: '2020-10-06T07:00:00.000Z',
      endsAt: '2020-10-06T13:00:00.000Z',
    },
    {
      id: 166,
      position: { name: 'Sports', color: 'lime', id: 233 },
      startsAt: '2020-07-14T10:00:00.000Z',
      endsAt: '2020-07-14T14:00:00.000Z',
    },
    {
      id: 195,
      position: { name: 'Sports', color: 'lime', id: 45 },
      startsAt: '2020-11-10T07:00:00.000Z',
      endsAt: '2020-11-10T13:00:00.000Z',
    },
    {
      id: 239,
      position: { name: 'Sports', color: 'lime', id: 45 },
      startsAt: '2020-02-05T06:00:00.000Z',
      endsAt: '2020-02-05T13:00:00.000Z',
    },
  ],
  pagination: { offset: 0, limit: 10, count: 9 },
};

export const EventDetails = {
  id: 95,
  startsAt: '2020-06-09T09:00:00.000Z',
  endsAt: '2020-06-09T13:00:00.000Z',
  position: { name: 'Sports', color: 'lime', id: 45 },
  employees: [
    {
      firstName: 'Bettye',
      lastName: 'Mann',
      image: 'https://cdn.fakercloud.com/avatars/vivekprvr_128.jpg',
      id: 73,
    },
  ],
};
