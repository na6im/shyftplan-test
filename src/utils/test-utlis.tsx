// test-utils.js
import React, { ComponentType, ReactChild, ReactChildren } from 'react';
import { render } from '@testing-library/react';
import { ToastProvider } from 'react-toast-notifications';
import { QueryClient, QueryClientProvider } from 'react-query';

interface AuxProps {
  children: ReactChild | ReactChildren;
}

const queryClient = new QueryClient();

function renderWithTheme(ui: React.ReactElement) {
  const Wrapper = ({ children }: AuxProps) => (
    <QueryClientProvider client={queryClient}>
      <ToastProvider>
        {children}
      </ToastProvider>
    </QueryClientProvider>
  );
  return {
    ...render(ui, { wrapper: Wrapper as ComponentType }),
  };
}

// override render method

export * from '@testing-library/react';

export { renderWithTheme };
