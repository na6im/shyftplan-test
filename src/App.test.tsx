import {eventsData} from "./__mocks__/mockedData";
import {renderWithQuery, screen} from "./utils/test-utlis";
import App from "./screens/EventsScreen";
import React from "react";
import mockedAxios from './__mocks__/axios';

it('Should renderWithQuery and match the snapshot', async () => {
    mockedAxios.get.mockImplementationOnce(() =>
        Promise.resolve({
            data: eventsData,
        })
    );
    const {
        container: { firstChild },
    } = renderWithQuery(<App />);

    expect(await screen.findByTestId(/container/i)).toBeInTheDocument();
    expect(firstChild).toMatchSnapshot();
});
