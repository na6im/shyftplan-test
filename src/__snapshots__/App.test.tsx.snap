// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should renderWithQuery and match the snapshot 1`] = `
<div
  class="sc-eCApnc cCbeHS"
  data-testid="isLoading"
>
  <img
    class="sc-fujyAs bOmVKR"
    src="logo-only.png"
  />
</div>
`;
